modular_skyrat/modules/modular_items/lewd_items/code/lewd_clothing/shibari_worn.dm

```diff
@@ -0,0 +1,268 @@
+		ropes.update_icon_state()
+		ropes.update_icon()
+	. = ..()
+
+/obj/item/clothing/under/shibari/update_icon_state()
+	. = ..()
+	icon_state = "[initial(icon_state)]_[current_color]"
+	worn_icon_state = "[initial(icon_state)]_[current_color]"
+
+/obj/item/clothing/under/shibari/ComponentInitialize()
+	. = ..()
+	AddElement(/datum/element/update_icon_updates_onmob)
+
+/obj/item/clothing/under/shibari/Initialize()
+	. = ..()
+	update_appearance()
+
+/obj/item/clothing/under/shibari/attack_hand(mob/user)
+	. = ..()
+	if(.)
+		return
+	if(iscarbon(user))
+		var/mob/living/carbon/human/hooman = user
+		if(src == hooman.w_uniform)
+			if(do_after(hooman, HAS_TRAIT(hooman, TRAIT_RIGGER) ? 2 SECONDS : 10 SECONDS, target = src))
+				qdel(src)
+		else
+			return
+
+
+//stuff to apply processing on equip and add mood event for perverts
+/obj/item/clothing/under/shibari/equipped(mob/user, slot)
+	.=..()
+	var/mob/living/carbon/human/hooman = user
+	if(src == hooman.w_uniform)
+		START_PROCESSING(SSobj, src)
+	if(HAS_TRAIT(hooman, TRAIT_ROPEBUNNY))
+		hooman.apply_status_effect(/datum/status_effect/ropebunny)
+
+//same stuff as above but for dropping item
+/obj/item/clothing/under/shibari/dropped(mob/user, slot)
+	.=..()
+	STOP_PROCESSING(SSobj, src)
+	var/mob/living/carbon/human/hooman = user
+	if(HAS_TRAIT(hooman, TRAIT_ROPEBUNNY))
+		hooman.remove_status_effect(/datum/status_effect/ropebunny)
+
+/obj/item/clothing/under/shibari/torso
+	name = "Shibari ropes"
+	desc = "Nice looking rope bondage."
+	icon_state = "shibari_body"
+
+//processing stuff
+/obj/item/clothing/under/shibari/torso/process(delta_time)
+	var/mob/living/carbon/human/hooman = loc
+	if(tight == "low" && hooman.arousal < 15)
+		hooman.adjustArousal(0.6 * delta_time)
+	if(tight == "medium" && hooman.arousal < 25)
+		hooman.adjustArousal(0.6 * delta_time)
+	if(tight == "hard" && hooman.arousal < 30)
+		hooman.adjustArousal(0.6 * delta_time)
+		if(hooman.pain < 25)
+			hooman.adjustPain(0.6 * delta_time)
+		if(prob(10))
+			hooman.adjustOxyLoss(5)
+
+/obj/item/clothing/under/shibari/groin
+	name = "Crotch rope shibari"
+	desc = "A rope that teases the wearer's genitals."
+	icon_state = "shibari_groin"
+
+//stuff to apply processing on equip and add mood event for perverts
+/obj/item/clothing/under/shibari/groin/equipped(mob/user, slot)
+	var/mob/living/carbon/human/hooman = user
+	if(hooman?.dna?.mutant_bodyparts["taur"])
+		slowdown = 4
+	else
+		slowdown = 0
+	. = ..()
+
+//processing stuff
+/obj/item/clothing/under/shibari/groin/process(delta_time)
+	var/mob/living/carbon/human/hooman = loc
+	if(tight == "low" && hooman.arousal < 20)
+		hooman.adjustArousal(0.6 * delta_time)
+		hooman.adjustPleasure(0.6 * delta_time)
+	if(tight == "medium" && hooman.arousal < 40)
+		hooman.adjustArousal(0.6 * delta_time)
+		hooman.adjustPleasure(0.6 * delta_time)
+	if(tight == "hard" && hooman.arousal < 60)
+		hooman.adjustArousal(0.6 * delta_time)
+		hooman.adjustPleasure(0.6 * delta_time)
```

Comment:
Same as previous comment, put them in one place, if you need to put the same exact code at more than one spot in your addition, then you need to rethink how you're doing it to reduce duplication of code.